src/v/kafka/server/tests/produce_consume_test.cc

```diff
@@ -159,3 +160,21 @@ FIXTURE_TEST(test_produce_consume_small_batches, prod_consume_fixture) {
       resp_2.data.topics.begin()->partitions.begin()->records->last_offset(),
       offset_2);
 };
+
+FIXTURE_TEST(test_version_handler, prod_consume_fixture) {
+    wait_for_controller_leadership().get();
+    start();
+    std::vector<kafka::produce_request::topic> topics;
+    topics.push_back(kafka::produce_request::topic{
+      .name = model::topic{"abc123"}, .partitions = small_batches(10)});
+
+    const auto unsupported_version = kafka::api_version(
+      kafka::produce_handler::max_supported() + 1);
+    BOOST_CHECK_THROW(
+      producer
+        ->dispatch(
+          kafka::produce_request(std::nullopt, 1, std::move(topics)),
+          unsupported_version)
+        .get(),
+      kafka::client::kafka_request_disconnected_exception);
+}
```

Comment:
do we want test for api_versions too that shows that for api_versions there is no range enforced?

Reply:
Yup, that already [exists](https://github.com/redpanda-data/redpanda/blob/dev/src/v/kafka/server/tests/api_versions_test.cc#L49) its a unit test that makes a test with the version set to `std::numeric_limits<kafka::api_version::type>`